src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

```diff
@@ -30,6 +30,7 @@
  *
  * @author Oliver Burn
  * @author <a href="mailto:dobratzp@ele.uri.edu">Peter Dobratz</a>
+ * @noinspection ClassWithTooManyDependents
```

Comment:
Why is this needed now? We didn't make any changes to this class, except documentation.

Reply:
Because TokenTypesDoclet depends on TokenTypes.
Any reference to the TokenTypes class leads to the violation "TokenTypes class has too many dependencies (301> 300)".
Roman suggest to suppress it: https://github.com/checkstyle/checkstyle/pull/5312#issuecomment-349306661

Reply:
ok. This is a weird violation where other classes cause a violation in the main class.